File: back/src/main/java/com/back/domain/review/validation/ValidRating.java
package com.back.domain.review.validation;

import jakarta.validation.Constraint;
import jakarta.validation.Payload;

import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.FIELD;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Target(FIELD)
@Retention(RUNTIME)
@Constraint(validatedBy = ValidRatingValidator.class)
public @interface ValidRating {
    String message() default "rating은 0~5, 0.5 단위여야 합니다.";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};
}

File: back/src/main/java/com/back/domain/review/validation/ValidRating.kt
package com.back.domain.review.validation

import jakarta.validation.Constraint
import jakarta.validation.Payload
import kotlin.reflect.KClass

@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
@Constraint(validatedBy = [ValidRatingValidator::class])
annotation class ValidRating(
    val message: String = "rating은 0~5, 0.5 단위여야 합니다.",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = []
)